docker/partition_gen.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+import sqlite3
+import multiprocessing
+from collections import defaultdict
+import json
+from math import ceil
+from pathlib import Path
+import sys
+
+
+def create_partitions(gpkg_path: Path, num_partitions: int = None) -> None:
+    if num_partitions is None:
+        num_partitions = multiprocessing.cpu_count()
+
+    con = sqlite3.connect(gpkg_path)
+    nexus = defaultdict(list)
+    for row in con.execute("SELECT toid, divide_id FROM divides"):
+        nexus[row[0]].append(row[1])
+    con.close()
+    num_partitions = min(num_partitions, len(nexus))
+    partition_size = ceil(len(nexus) / num_partitions)
+    num_nexus = len(nexus)
+    nexus = list(nexus.items())
+    partitions = []
+    for i in range(0, num_nexus, partition_size):
+        part = {}
+        part["id"] = i // partition_size
+        part["cat-ids"] = []
+        part["nex-ids"] = []
+        part["remote-connections"] = []
```

Comment:
For any nexus that falls on a partition boundary, this remote connection becomes a vital price of information to the model engine. If it's always empty, I doubt you will get the correct simulation results in anything other than the most trivial domains.

Reply:
only 95% sure I did my original testing on the latest next gen in a box so I need to go back and test it again, but this exists because I needed to quickly fix an issue and don't properly understand remotes.
 I'll generate examples in the afternoon after finals, but the gist of it was that when I partitioned with dmod, it would split the catchments equally with 1 catchment per partition, then add the nexus the catchment flowed to, which resulted in nexuses being duplicated over partitions in cases where more than one catchment flowed into it. Then when running this in parallel, multiple processes were appending to the same file and corrupting it. Reversing the partitioning to be one per nexus and then adding the catchments stopped the files corrupting.

Reply:
not sure what versions I used that caused this issue, but the latest NGIAB has no issues with the file corruption so I can delete all of this python partitioning and use the dmod partitioning. 
The only modification needed is a check in the command to call the partition_generator to make sure that I'm not trying to partition 5 catchments into nproc partitions and instead use at mosts 5.